### PR TITLE
Add suggestion commands

### LINKED ISSUE #113: Create Suggestion Commands classes
To create suggestion commands classes that implement `SuggestionCommand` interface. For v1.2, will limit to just these 2 classes:

- `OpenSuggestionCommand`: when user types `"open ..."`
- `SearchSuggestionCommand`: when user types `"search ..."`

`logic --> suggestion --> commands --> OpenSuggestionCommand.java`
`logic --> suggestion --> commands --> SearchSuggestionCommand.java`

### PR DESCRIPTION
**Pull request by johannagwan**
Closes #113 
Currently added:
- OpenSuggestionCommand
- SearchSuggestionCommand
- ErrorSuggestionCommand
- SuggestionListStub (just for the sake of testing, will be deleted in the future)

When a particular suggestion is chosen, the `commandInputText` in the model will be updated.
Currently, I just used a hard-coded list for the data.

### CODE REVIEW

File: src/main/java/com/notably/logic/suggestion/commands/ErrorSuggestionCommand.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Represents an error command when the user input do not match any of the available commands or blocks.
+ */
+public class ErrorSuggestionCommand implements SuggestionCommand {
+    @Override
+    public void execute(Model model) {
+
+    }
+}
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/commands/ErrorSuggestionCommand.java:13]: Any further implementation?

File: src/main/java/com/notably/logic/suggestion/commands/OpenSuggestionCommand.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import com.notably.logic.suggestion.stubs.SuggestionListStub;
+import com.notably.model.Model;
+import com.notably.model.suggestion.SuggestionItem;
+import com.notably.model.suggestion.SuggestionItemImpl;
+
+/**
+ * Represents a suggestion command object to open a note.
+ */
+public class OpenSuggestionCommand implements SuggestionCommand {
+    private Path path;
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/commands/OpenSuggestionCommand.java:16]: We're deprecating `Path`. @ljiazh3ng is working on a new set of `AbsolutePath` and `RelativePath` in #116 . To be merged soon!

Please keep a watch for it :)

File: src/main/java/com/notably/logic/suggestion/commands/SearchSuggestionCommand.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        model.setSuggestions(suggestions);
+    }
+
+    private List<String> getSuggestionList(Path path) {
+        List<String> paths = path.getComponents();
+
+        // TODO: traverse tree. For now will just use stubs.
+        SuggestionListStub suggestionListStub = new SuggestionListStub();
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/commands/SearchSuggestionCommand.java:43]: Oh this is not how stubs are supposed to be used!

For your case, perhaps you can just hardcode a list of `String`s here in this class.

**Review by kevinputera — COMMENTED** [review]
Hey @johannagwan , great job here!

Some suggestions:
1. Next time, break down your changes into even smaller subparts! You can create a separate PR for each of these commands.
2. Your `getSuggestionList` private method could use some refactoring/renaming to increase cohesiveness! Some possible improvements:
   1. Rename `getSuggestionList` method to `getPossiblePaths` and return `List<Path>`.
   2. Rename `getSuggestionList` method to `getSuggestions` and return `List<SuggestionItem>`

Let's not merge this in first, at least until the reworking of the `Path` (#116) object is done!

**Comment by kevinputera** [comment]
@johannagwan I haven't really looked through the exact workings of both `SearchSuggestionCommand` and `OpenSuggestionCommand` yet since there is a chance their implementation might change after `AbsolutePath` and `RelativePath` are introduced. Sorry about that!

**Comment by kevinputera** [comment]
@johannagwan Another thing, please add an issue relevant to this PR to explain more about the changes proposed! At the end of the day, it helps you think through your implementation and also gets accounted for by the grading bot :)

**Comment by johannagwan** [comment]
Hi @kevinputera thanks for the review! Is it better for me to just close this PR and issue #113 then create new issues and new PRs, each for the suggestion commands?

**Comment by kevinputera** [comment]
> Hi @kevinputera thanks for the review! Is it better for me to just close this PR and issue #113 then create new issues and new PRs, each for the suggestion commands?

Not really necessary, but up to you!

**Pull request closed without merging**